salt/matchers/glob_match.py

```diff
@@ -15,7 +15,7 @@ def match(tgt, opts=None, minion_id=None):
     if not opts:
         opts = __opts__
     if not minion_id:
-        minion_id = opts.get("minion_id", opts["id"])
```

Comment:
`minion_id` was kept in the code base because @garethgreenaway mentioned he found reference to it in pillar, so he did not want to make a breaking change.

Reply:
For reference: https://github.com/saltstack/salt/pull/59603#discussion_r585741303

Reply:
Hmm. Ok. I'll add that back.

Reply:
Let's merge this and I'll make a follow-up PR adding it back.

Reply:
@s0undt3ch will take care of this in another PR and we will merge here to get tests passing.

Reply:
https://github.com/saltstack/salt/pull/59723